ios/Classes/E3dbSerializer.swift

```diff
@@ -0,0 +1,64 @@
+import E3db
+
+public class E3dbSerializer {
+    static func realmFromJson(json: String) -> Application {
+        let jsonData = json.data(using: .utf8)!
+        let realmConfig: RealmConfigSerializer = try! JSONDecoder().decode(RealmConfigSerializer.self, from: jsonData)
+        return Application(apiUrl: realmConfig.apiURL, appName: realmConfig.appName, realmName: realmConfig.realmName, brokerTargetUrl: realmConfig.brokenTargetURL)
+    }
+    
+    static func idClientToJson(id: Identity) -> String {
+        var data: [String: Any] = [:]
+
+        data.updateValue(id.idConfig.storageConfig, forKey: "client_credentials")
+        data.updateValue(E3dbSerializer.idConfigToJson(idConfig: id.idConfig), forKey: "identity_config")
+        data.updateValue(E3dbSerializer.userAgentTokenToJson(tok: id.agentInfo()), forKey: "user_agent_token")
+        
+        let jsonData = try! JSONSerialization.data(withJSONObject: data, options: .prettyPrinted)
+        guard let jsonString = String(data: jsonData, encoding: .utf8) else { return "Something bad happened." } // TODO: Error
```

Comment:
Let's update these "Something bad happened" to "Failed to <what failed in 3-7 words>" so for this something like `Failed to stringify an idClient JsonObject`. 

Also as I am not a swift expert would it be better to have this throw an error of some kind if this method fails? With this string a downstream consumer will need to equality check for failures, is there a more canonical way to show failures to method callers? Potentially that is what the error handling notes you have are about, if so we should do that sooner rather than later.

Reply:
Yup, that is the intention of the error handling notes. An option is to, instead of `guard` (which returns early), provide a default value. e.g. `let jsonString = String(...) ?? ""`. There are better patterns where I can throw errors, though. The `try!` is a forced unwrapping, which is generally discouraged. Updating the error handling will include removing forced unwraps 

I agree the guard pattern is not what should be used here, these will be fixed.

Reply:
We should return some type of error whether it is checked and handled or a runtime exception that crashes the program with a serialization error (That appears to be the approach we use on the java side). But propagating a string that downstream functions will not be able to correctly handle seems like the worst of the three options